token-swap/program/src/curve/calculator.rs

```diff
@@ -99,12 +102,14 @@ pub trait CurveCalculator: Debug + DynPack {
         swap_token_a_amount: u128,
         swap_token_b_amount: u128,
     ) -> Option<TradingTokenResult> {
-        let token_a_amount = pool_tokens
-            .checked_mul(swap_token_a_amount)?
-            .checked_div(pool_token_supply)?;
-        let token_b_amount = pool_tokens
-            .checked_mul(swap_token_b_amount)?
-            .checked_div(pool_token_supply)?;
+        let (token_a_amount, _) = ceiling_division(
```

Comment:
Should ceiling division be applied to the other `pool_tokens_to_trading_tokens` methods too?

Reply:
Yes absolutely, great catch